python/dolfinx/fem/element.py

```diff
@@ -46,6 +46,55 @@ def dtype(self) -> np.dtype:
         """Scalar type for the coordinate element."""
         return np.dtype(self._cpp_object.dtype)
 
+    @property
+    def dim(self) -> int:
+        """The dimension of the coordinate element space.
+
+        The number of basis function is returned. e.g., for a linear
+        triangle cell the dimension will be 3.
+        """
+        return self._cpp_object.dim
+
+    def create_dof_layout(self) -> _cpp.fem.ElementDofLayout:
+        """Compute and return the dof layout"""
+        return self._cpp_object.create_dof_layout()
+
+    def push_forward(self, X: np.ndarray, cell_geometry: np.ndarray) -> np.ndarray:
+        """Compute the physical coordinates ``x`` of the reference coordinates ``X``.
+
+        Args:
+            X: The physical coordinates of the reference points,
+                shape ``(num_points, topological_dimension)``.
+            cell_geometry: Physical coordinates describing the cell,
```

Comment:
It would be helpful to describe `cell_geometry`. I'm not sure from the docstring what should be passed in.

Reply:
I am not sure what you would like to add here. 
It is the physical nodes/dofs of a given cell of the coordinate element.

Reply:
Could could say something like 'geometry degree-of-freedom` and how the user would access the data in the right format.